Sources/App/Core/SearchLogger.swift

```diff
@@ -25,18 +37,32 @@ enum SearchLogger {
 //        #endif
 //    }
 
+
     static func log(query: String, results: [Search.Result]) {
-        struct SearchFragment {
-            let searchID: UUID
-            let query: String
-            let result: Search.Result
-        }
         
         let uniqueSearchID = UUID()
+        let jsonEncoder = JSONEncoder()
+        jsonEncoder.dateEncodingStrategy = .iso8601
+        jsonEncoder.outputFormatting = .sortedKeys
 
-        for result in results {
-            let fragment = SearchFragment(searchID: uniqueSearchID, query: query, result: result)
-            AppEnvironment.logger.info("searchfragment: \(fragment)")
+        if results.isEmpty {
+            let fragment = SearchFragment(searchID: uniqueSearchID, query: query, result: nil)
+            do {
+                let stringdata = try jsonEncoder.encode(fragment)
+                AppEnvironment.logger.info("searchfragment: \(fragment)")
+            } catch {
+                AppEnvironment.logger.warning("unable to encode search fragment: \(error)")
+            }
+        } else {
+            for result in results {
+                let fragment = SearchFragment(searchID: uniqueSearchID, query: query, result: result)
```

Comment:
I'm pretty sure you'll want to enumerate the results and add an index field to `SearchFragment` in order to preserve order. It should be logged in order given a search will happen on a single node but it's probably better to fully embed that info.

Reply:
Both excellent suggestions - and I'll aim to do them both. I'm about to head on a flight though, so this may get stalled a bit through the new year. You guys _should_ definitely get a holiday break as well. I'll try and get something updated in a few hours, but if I miss today - it'll be mid-january before I'm back in front of a computer to do anything useful (including fixing the embedded ugliness in the SPISearch app)

Reply:
Have a safe trip! :)